### PR TITLE
Several fixups for issues found by Coverity

### LINKED ISSUE #27: 4 new defect(s) introduced to dfuzzer found with Coverity Scan
https://scan.coverity.com/projects/dfuzzer

On the bright side "7 defect(s), reported by Coverity Scan earlier, were marked fixed in the recent build analyzed by Coverity Scan."

### PR DESCRIPTION
**Pull request by mrc0mmand**
Fixes #27 (and more).

### CODE REVIEW

**Comment by evverx** [comment]
@mrc0mmand in https://github.com/matusmarhefka/dfuzzer/pull/26 I switched the GHActions CI to meson and there `dfuzzer` is built with `gcc/clang`. I think it would sense to merge that PR and rebase this PR on top of it to test it with both clang and gcc. `meson` is used on GitHub only and it doesn't affect the package so it should be safe to merge that PR

File: src/dfuzzer.c [hunk 1]
```diff
@@ -854,6 +852,7 @@ void df_print_process_info(int pid)
                 return;
         }
 
+        memset(name, 0, PATH_MAX);
```
Comment (evverx) [inline src/dfuzzer.c:855]: I'd replace it with `explicit_bzero`. Having said that I need to take a look at the coverity report to figure out what this is about.

**Comment by evverx** [comment]
CID#254663 seems to be a false positive in the sense that apparently Coverity thinks that `(name + i)` and `name[i]` have nothing to do with each other in
```c
ret = read(fd, (name + i), 1);
...
if (name[i] == '\0')
    break
```
Having said that if code confuses Coverity it usually means that it's hard to follow in general and should be rewritten. I think it would make sense to drop `explicit_bzero` there for now.

**Comment by mrc0mmand** [comment]
> CID#254663 seems to be a false positive in the sense that apparently Coverity thinks that `(name + i)` and `name[i]` have nothing to do with each other in
> 
> ```c
> ret = read(fd, (name + i), 1);
> ...
> if (name[i] == '\0')
>     break
> ```
> 
> Having said that if code confuses Coverity it usually means that it's hard to follow in general and should be rewritten. I think it would make sense to drop `explicit_bzero` there for now.

I see, that makes sense, thanks for digging into that.
